### PR TITLE
Handle faulty v2_key while upgrading ansible

### PR DESCRIPTION
**Pull request by kbrock**
Situation:

When importing a foreign database, the v2_key and therefore the
ansible key is not available. Which is good/ok since we shouldn't
be able to read a customer's passwords anyway.

What is bad is there is no way to work with this scenario. The database
simply can not be migrated.

Solution:

Provide a way to continue even though we can't read all the ansible
passwords.

run migration with env variable to just get bogus passwords

INVALID_ANSIBLE_VALUE=bogus rake db:migrate

original PR:
https://github.com/ManageIQ/manageiq-schema/pull/385

https://bugzilla.redhat.com/show_bug.cgi?id=1755553

### CODE REVIEW

**Comment by kbrock** [comment]
Nick, I need some help with the specs here.

@miq-bot add_reviewer NickLaMuro

File: db/migrate/20190617191109_move_awx_credentials_to_authentications.rb [hunk 1]
```diff
@@ -267,6 +270,15 @@ def up
...
   rescue PG::ConnectionBad
     say("awx database is unreachable, credentials cannot be migrated")
+  rescue Fernet256::InvalidToken
+    say("awx password invalid, credentials cannot be migrated")
+    say("This is often the case when migrating a support database")
+    say("And fix_auth has reset the ansible secret key")
+    say("")
+    say("If you want to continue and set all the invalid passwords")
+    say("to a preknown bogus value then")
+    say("run migrate again with $INVALID_ANSIBLE_VALUE=bogus"
```
Comment (NickLaMuro) [inline db/migrate/20190617191109_move_awx_credentials_to_authentications.rb]: Looks like you are short a `)` at the end here.

File: db/migrate/20190617191109_move_awx_credentials_to_authentications.rb [hunk 2]
```diff
@@ -139,6 +139,9 @@ def self.decrypt(field_name, value, primary_key)
       return value unless value.include?("$encrypted$")
 
       new(secret_key, value, field_name, primary_key).decrypt
+    rescue Fernet256::InvalidToken
```
Comment (jrafanie) [inline db/migrate/20190617191109_move_awx_credentials_to_authentications.rb]: Is this a very specific class exception?  Is there a higher level invalid token class we can use?

Comment (NickLaMuro) [reply db/migrate/20190617191109_move_awx_credentials_to_authentications.rb]: @jrafanie look higher up in the file 😉 

https://github.com/ManageIQ/manageiq-schema/blob/a3a2dcfe7cd63e65b3b7ba70634247519183408e/db/migrate/20190617191109_move_awx_credentials_to_authentications.rb#L23

File: db/migrate/20190617191109_move_awx_credentials_to_authentications.rb [hunk 3]
```diff
@@ -139,6 +139,9 @@ def self.decrypt(field_name, value, primary_key)
...
+      raise unless ENV["INVALID_ANSIBLE_VALUE"]
+      ENV["INVALID_ANSIBLE_VALUE"]
```
Comment (jrafanie) [inline db/migrate/20190617191109_move_awx_credentials_to_authentications.rb]: Does this value need to be known later for fix_auth?  Can it be any value?

Words are hard.  Maybe HARDCODE_ANSIBLE_PASSWORD or something like that?

File: db/migrate/20190617191109_move_awx_credentials_to_authentications.rb (hunk 2 above)
Comment (jrafanie) [reply db/migrate/20190617191109_move_awx_credentials_to_authentications.rb]: Oh wow, I forgot we have that implemented here.  👍

**Comment by jrafanie** [comment]
This workflow seems ok to me.  I'm not sure how to word this but it seems reasonable to get a database dump from someone and not caring about the actual ansible password, you just want to migrate the DB using a fake or hardcoded invalid password.

**Comment by NickLaMuro** [comment]
@jrafanie @kbrock For simplicity, I created a new version of this PR with the specs added:

https://github.com/ManageIQ/manageiq-schema/pull/419

I ended up rebased my specs into the original commit, so didn't think that merging my changes into this PR would work out.  @kbrock should still have credit as the author of the commit though since it was the one that I merged into.

Hope that is okay.

File: db/migrate/20190617191109_move_awx_credentials_to_authentications.rb (hunk 3 above)
Comment (kbrock) [reply db/migrate/20190617191109_move_awx_credentials_to_authentications.rb]: can be any value. In fixauth, we typically choose `"bogus"`. But it need not be the same as fix_auth

**Pull request closed without merging**